core/src/subgraph/runner.rs

```diff
@@ -144,12 +144,59 @@ where
 
             let block_stream_canceler = CancelGuard::new();
             let block_stream_cancel_handle = block_stream_canceler.handle();
+            let current_ptr = self.inputs.store.block_ptr();
+
+            let static_filters = self.inputs.static_filters
+                || self.ctx.instance().hosts().len() > ENV_VARS.static_filters_threshold;
+
+            // if static_filters is enabled, build a minimal filter with the static data sources and
+            // add the necessary filters based on templates.
+            // if not enabled we just stick to the filter based on all the data sources.
+            // This specifically removes dynamic data sources based filters because these can be derived
+            // from templates AND this reduces the cost of egress traffic by making the payloads smaller.
+            let filter = if static_filters {
```

Comment:
Why was this moved to the runner?

Reply:
We now need to rebuild the TriggerFilter every time the Blockstream is restarted, this is because  its not possible to remove a datasource from TriggerFilter when its endBlock is reached, As per my discussion with @leoyvens we decided reconstructing the trigger filter is a more robust way to deal with this